src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -3011,5 +3014,108 @@ class C
                 state.AssertItemsInOrder(expectedOrder)
             End Using
         End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function TestLargeChangeBrokenUpIntoSmallTextChanges() As Task
+            Using state = TestState.CreateCSharpTestState(
+                <Document><![CDATA[
+using System;
+class C
+{
+    void foo() {
+        return$$
```

Comment:
:bulb: It would be good to have a test which uses the following code (or equivalent):

```
class C
{
  void foo() {
    f$$oo();
  }
}
```

With completion invoked at the `$$` location, no actual edits are made but the caret moves to after `foo`.

A related scenario would be the following:

```csharp
throw new NotImExc$$eption();
```

In this scenario, edits are made but the edits do not end at the same location where the caret ends up.